pkg/framework/context/context.go

```diff
@@ -129,14 +129,16 @@ func (p *Provider) InboundMessageHandler() transport.InboundMessageHandler {
 		// in case of no services are registered for given message type,
 		// find generic inbound services registered for given message header
 		for _, svc := range p.msgSvcProvider.Services() {
-			h := &service.Header{}
```

Comment:
you should remember to delete purpose from header.
https://github.com/hyperledger/aries-framework-go/blob/4de0955d13d7123da31c2f2386f4d3fb25f3106b/pkg/didcomm/common/service/did_comm_msg.go#L35

Reply:
whole `service.header` going to be removed, I removed it's reference from context and message service in this PR.

Reply:
Is the next PR to remove the whole header?

Reply:
Removed `Purpose` now and either me or @soluchok will work on #1075 to remove deprecated header structure.